lib/bencode.ml

```diff
@@ -62,6 +62,49 @@ let rec pretty_print =
           Buffer.add_string buf ";\n")
   in loop 1
 
+let pp (fmt:Format.formatter) (benc:t) =
+  let pp_nest fmt nestlvl = (* prints a comment explaining tree context *)
+    if nestlvl = [] then ()
+    else Format.fprintf fmt " (* %a *)"
+        Format.(pp_print_list
+                  ~pp_sep:(fun fmt () -> pp_print_string fmt " -> ")
+                  (fun fmt str -> pp_print_string fmt (String.escaped str))
+               ) ("^--" :: List.rev nestlvl)
+  in
+  let rec pp_val (nestlvl:string list) fmt = function
+    | Integer x -> Format.pp_print_string fmt @@ Int64.to_string x
```

Comment:
you can use `Format.fprint fmt "%Ld" x`, but I'm not even sure if it's faster ^^

Reply:
No idea whatsoever what is faster, but you're right that `fprintf` is easier to read, will change it to that.